File: examples/javascript/implementation.js
export const Logger = {
  log: (message) => console.log(message),
};

export class NotNumberError extends Error {
  constructor(message = "Not a number!") {
    super(message);
  }
}

export const calculate = (x, y) => {
  if (typeof x !== "number" || Number.isNaN(x)) {
    throw new NotNumberError(
      `x is not a number or is NaN. x: ${typeof x} = ${x}`
    );
  }

  if (typeof y !== "number" || Number.isNaN(y)) {
    throw new NotNumberError(
      `y is not a number or is NaN. y: ${typeof y} = ${y}`
    );
  }

  Logger.log(`Calculating for (x, y) = (${x}, ${y})...`);

  return x + y;
};

File: examples/implementation.py
class NotNumberError(Exception):
    def __init__(self, message="Not a number!"):
        super().__init__(message)


class Logger:
    @staticmethod
    def log(message):
        print(message)


def calculate(x, y):
    if not isinstance(x, (int, float)) or x != x:  # x != x checks for NaN
        raise NotNumberError(f"x is not a number or is NaN. x: {type(x)} = {x}")
    if not isinstance(y, (int, float)) or y != y:  # y != y checks for NaN
        raise NotNumberError(f"y is not a number or is NaN. y: {type(y)} = {y}")
    Logger.log(f"Calculating for (x, y) = ({x}, {y})...")
    return x + y